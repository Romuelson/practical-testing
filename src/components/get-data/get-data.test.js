const axios = require('axios');

const getData = require('./get-data');
const mockUserData = require('../../mocks/users-data');

jest.mock('axios');

describe('Компонент: getData', () => {
	let response;

	beforeEach(() => {
		response = {
			data: mockUserData
		};
	});

	it('Произведет успешный вызов и вернет преобразованные данные', async () => {
		axios.get.mockReturnValue(response);

		const data = await getData();

		expect(axios.get).toBeCalledTimes(1);
		expect(data).toEqual(['1', '2', '3']);
		expect(data).toMatchSnapshot(['1', '2', '3']);
	});
});
