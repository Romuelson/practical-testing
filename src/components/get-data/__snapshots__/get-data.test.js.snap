// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент: getData Произведет успешный вызов и вернет преобразованные данные 1`] = `
Array [
  "1",
  "2",
  "3",
]
`;
